MetaMask/metamask-extension
# fix: Add L2 support in ConfirmLegacyGasDisplay to support Scroll Network transactions

## **Description**

Scroll supports legacy transactions and does not support EIP-1559. This PR adds L2 support in the ConfirmLegacyGasDisplay component

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/24854?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24835
Follow-up Cherry-pick: https://github.com/MetaMask/metamask-extension/pull/24866
## **Manual testing steps**

1. Go to test dapp
2. Click on "Create a token"
3. Notice the gas component and its values in each different version/build


## **Screenshots/Recordings**



### **Before**

![Screenshot 2024-05-28 at 14 01 09](https://github.com/MetaMask/metamask-extension/assets/20778143/9e8eb8ea-801a-4759-8394-0293eb6565d1)


### **After** (rate limited so $$ fiat amounts are not shown in After screenshot)

<img src="https://github.com/MetaMask/metamask-extension/assets/20778143/f21d5438-8be4-42c9-a892-9e8782cff565" width="380">

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
